tests/test_io_utils.py

```diff
@@ -0,0 +1,57 @@
+import json
+import os
+
+import fsspec
+import pandas as pd
+
+import ecoscope
+
+
+def test_download_file_github_csv():
+    ECOSCOPE_RAW = "https://raw.githubusercontent.com/wildlife-dynamics/ecoscope/master"
+    output_dir = "tests/test_output"
+    ecoscope.io.download_file(
+        f"{ECOSCOPE_RAW}/tests/sample_data/vector/movbank_data.csv",
+        os.path.join(output_dir, "download_data.csv"),
+        overwrite_existing=True,
+    )
+
+    data = pd.read_csv(os.path.join(output_dir, "download_data.csv"))
+    assert len(data) > 0
+
+
+def test_download_file_gdrive():
+    output_dir = "tests/test_output"
+    ecoscope.io.download_file(
+        "https://drive.google.com/uc?export=download&id=1-AQ9_oacUCcAaiZ6SWU77hZWp1oArQw6",
+        os.path.join(output_dir, "download_data.csv"),
+        overwrite_existing=True,
+    )
+
+    data = pd.read_csv(os.path.join(output_dir, "download_data.csv"))
+    assert len(data) > 0
+
+
+def test_download_file_gdrive_zip():
```

Comment:
Is it possible to combine with the above function so that it works for both non-zip and zip files?

Reply:
It works for both non-zip and zip files. Here are just two different unittests